### PR TITLE
homework week1

### PR DESCRIPTION
**Pull request by gisaac85**
All done!
============================
 √ /state returns 10
  √ /add returns 11
  √ /subtract returns 9
  √ /reset returns 10
  √ querying undefined URL returns 404 Not Found
  √ /add, /reset returns 10
  √ /subtract, /reset returns 10 (101ms)
  √ /add, /add, /add, /subtract returns 12 (128ms)
  √ /subtract, /subtract, /reset, /add, /subtract, /add returns 11 (150ms)
  √ /add, /add, /add, /add, /add, /add, /add, /add, /add, /add returns 20 (176ms)
  √ /subtract, /subtract, /subtract, /subtract, /subtract, /subtract, /subtract, /subtract, /subtract, /subtract returns 0 (179ms)

  11 tests passed

### CODE REVIEW

File: week1/homework/src/server.js [hunk 1]
```diff
@@ -10,11 +10,56 @@ function createServer(port) {
...
     // TODO: Write your homework code here
-  });
 
+    switch (request.url) {
+      case '/state':
+        state = 10;
+        let number = JSON.stringify({
+          state: state
+        });
+        response.setHeader('Content-Type', 'application/json');
```
Comment (GeorgeSapkin) [inline week1/homework/src/server.js]: You can also put the `setHeader` outside of the switch so you don't have to repeat it.

File: week1/homework/src/server.js [hunk 2]
```diff
@@ -10,11 +10,50 @@ function createServer(port) {
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
-  });
+    response.setHeader('Content-Type', 'application/json');
+    switch (request.url) {
+      case '/state':
+        state = 10;
```
Comment (alxbrg) [inline week1/homework/src/server.js]: You're re-assigning `10` to `state` here. Which means that this route actually does the same thing as the `/reset` one.

Comment (alxbrg) [inline week1/homework/src/server.js]: You're reassigning `10` to `state` here. That means that this route does the same thing as the `/reset` route, i.e. resetting `state` to `10`.

File: week1/homework/src/server.js [hunk 3]
```diff
@@ -10,11 +10,50 @@ function createServer(port) {
...
+        let number = JSON.stringify({
+          state: state
+        });
+        response.write(`${number}`);
```
Comment (alxbrg) [inline week1/homework/src/server.js]: You don't need to use a template string. `JSON.stringify` already return a `string`.

File: week1/homework/src/server.js (hunk 2 above)
Comment (GeorgeSapkin) [reply week1/homework/src/server.js]: Overlooked that one.

**Pull request closed without merging**